src/backend/distributed/commands/function.c

```diff
@@ -192,6 +202,22 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
 
+	if (IsCreateDistributedFunctionCallIdempotent(functionAddress,
+												  distributionArgumentName,
+												  colocateWithTableNameDefault,
+												  forceDelegationAddress))
+	{
+		char *schemaName = get_namespace_name(get_func_namespace(funcOid));
+		char *functionName = get_func_name(funcOid);
+		char *qualifiedName = quote_qualified_identifier(schemaName, functionName);
+		ereport(NOTICE, (errmsg("procedure %s is already distributed", qualifiedName),
+						 errdetail("Citus distributes procedures with CREATE commands"),
+						 errhint("To delegate the procedure, please provide parameters "
```

Comment:
Hmm, should we never give the hint? Because now we create a confusing message to the users who has no clue / not interested in the function delegation? Users are only interested in the delegation when colocateWithTableNameDefault != NULL and here it is definitely null?